code/modules/medical/surgery_procs.dm

```diff
@@ -633,19 +633,32 @@ var/global/list/chestitem_whitelist = list(/obj/item/gnomechompski, /obj/item/gn
 						if(surgeon == patient) continue
 						remove_mindslave_status(patient, "otherslave", "surgery")
 
-				patient.tri_message("<span class='alert'><b>[surgeon]</b> cuts out an implant from [patient == surgeon ? "[him_or_her(patient)]self" : "[patient]"] with [src]!</span>",\
-				surgeon, "<span class='alert'>You cut out an implant from [surgeon == patient ? "yourself" : "[patient]"] with [src]!</span>",\
-				patient, "<span class='alert'>[patient == surgeon ? "You cut" : "<b>[surgeon]</b> cuts"] out an implant from you with [src]!</span>")
-
-				var/obj/item/implantcase/newcase = new /obj/item/implantcase(patient.loc, usedimplant = I)
-				I.on_remove(patient)
-				patient.implant.Remove(I)
-				var/image/wadblood = image('icons/obj/surgery.dmi', icon_state = "implantpaper-blood")
-				wadblood.color = patient.blood_color
-				newcase.UpdateOverlays(wadblood, "blood")
-				newcase.blood_DNA = patient.bioHolder.Uid
-				newcase.blood_type = patient.bioHolder.bloodType
+				if (!I.artifact_implant)
+					patient.tri_message("<span class='alert'><b>[surgeon]</b> cuts out an implant from [patient == surgeon ? "[him_or_her(patient)]self" : "[patient]"] with [src]!</span>",\
+					surgeon, "<span class='alert'>You cut out an implant from [surgeon == patient ? "yourself" : "[patient]"] with [src]!</span>",\
+					patient, "<span class='alert'>[patient == surgeon ? "You cut" : "<b>[surgeon]</b> cuts"] out an implant from you with [src]!</span>")
+
+					var/obj/item/implantcase/newcase = new /obj/item/implantcase(patient.loc, usedimplant = I)
+					I.on_remove(patient)
+					patient.implant.Remove(I)
+					var/image/wadblood = image('icons/obj/surgery.dmi', icon_state = "implantpaper-blood")
+					wadblood.color = patient.blood_color
+					newcase.UpdateOverlays(wadblood, "blood")
+					newcase.blood_DNA = patient.bioHolder.Uid
+					newcase.blood_type = patient.bioHolder.bloodType
+				else
+					if (I.cant_take_out)
```

Comment:
As written, this makes scalpel organ surgeries impossible if you have an unremovable implant, since you'll just repeatedly try and fail to remove the implant. Not really an issue here because all the unremovable implants kill you in short order, but might come up in the future.